docs/root/faq/rolling_upgrades.rst

```diff
@@ -0,0 +1,124 @@
+and network failures. Having these features implemented at Envoy not only improves the availability 
+and resiliency of services but also brings in consistency in terms of the behaviour and observability. 
+
+This section explains at a high level the configuration supported by Envoy and how these features can be 
+used together to handle these scenarios.
+
+Circuit Breaking
+----------------
+
+:ref:`Circuit Breaking<arch_overview_circuit_break>` is a critical component of distributed systems. 
+Circuit breaking lets applications configure failure thresholds that ensure safe maximums, allows components
+to fail quickly and apply back pressure as soon as possible. Applying right circuit breaking thresholds helps
+to save resources which otherwise are wasted in waiting for requests (timeouts) or retrying requests unnecessarily. 
+One of the main advantages of circuit breaking implementation in Envoy is that the circuit breaking limits are applied
+at the network level.
+
+Retries
+-------
+
+Automatic :ref:`request retries<config_http_filters_router>` is another measure of ensuring service resilience. Request retries should typically be used to guard against 
+transient failures. Envoy supports very rich set of configurable parameters that dictate what type of requests are retried, how many times 
+the request should be retried, timeouts for retries, etc.
+
+Retries in gRPC services
+------------------------
+
+For gRPC services, Envoy looks at the gRPC status in the response and attempts a retry based on the statuses configured in *x-retry-grpc-on*.
+
+The following application status codes in gRPC are considered safe for automatic retry.
+
+* *CANCELLED* - Return this code if there is an error that can be retried in the service.
+* *RESOURCE_EXHAUSTED* - Return this code if some of the resources that service depends on are exhausted in that instance so that retrying 
+  to another instance would help. Please note that for shared resource exhaustion, returning this will not help. Instead :ref:`rate limiting<arch_overview_rate_limit>`
+  should be used to handle such cases.
+
+The Http Status codes *502 (Bad Gateway)*, *503 (Service Unavailable)* and *504 (Gateway Timeout)* are all mapped to gRPC status code *UNAVAILABLE*. 
+This can also be considered safe for automatic retry.
+
+An important consideration should be given to *idempotency* of a request while configuring retries.
+
+Envoy also supports various extensions to the standard retry policy supported out of the box. 
+The :ref:`retry plugins<arch_overview_http_retry_plugins>` allow you to customize the Envoy 
+retry implementation to your application.
+
+Outlier Detection
+-----------------
+
+:ref:`Outlier detection<arch_overview_outlier_detection>` is a way of dynamically determining outlier 
+hosts in the upstream cluster. If certain hosts behave differently from others like returning high 5xx 
```

Comment:
I would replace "determining outlier hosts" with "detecting misbehaving hosts" and remove the sentence starting with "If certain hosts". (Reads can click through to the outlier detection page for details on what misbehaving means.)